### PR TITLE
If we post to the inbox of a deleted user, send a 404 or 410 error.

### PR DESCRIPTION
**Pull request by KrinsKumar**
This PR addresses the issue https://github.com/activitypods/activitypods/issues/348

Few questions I had:
1. Currently I am only checking if a deletedAt field exists in an account to return 410, should I also check if that date is more than one year old and return 404 accordingly?
2. The issue said that this logic should not throw, but I implemented the changes using throws. Please let me know if this works.

### CODE REVIEW

File: src/middleware/packages/triplestore/actions/query.js
Comment (srosset81) [inline src/middleware/packages/triplestore/actions/query.js]: It's better not to include HTTP error codes here, as it is used directly by many services. If an account doesn't exist, the `auth.account.findByUsername` action that you called in the InboxService will return nothing, and so you can throw the 404 error from there.

File: src/middleware/packages/activitypub/services/activitypub/subservices/inbox.js
Comment (srosset81) [inline src/middleware/packages/activitypub/services/activitypub/subservices/inbox.js]: Please use the MoleculerError to include directly the HTTP code:
https://github.com/moleculerjs/moleculer/blob/4014a8c873956346c17357701433f0cc9c01be8e/src/errors.js#L53
For 404 error, there is a dedicated moleculer-web error:
https://github.com/moleculerjs/moleculer-web/blob/c6ec80056a64ea15c57d6e2b946ce978d673ae92/src/errors.js#L127

**Review by srosset81 — CHANGES_REQUESTED** [review]
Thanks !
Checking the `deletedAt` is enough. After one year, the account will be automatically deleted.

File: src/middleware/packages/triplestore/actions/query.js
Comment (KrinsKumar) [reply src/middleware/packages/triplestore/actions/query.js]: fixed.

File: src/middleware/packages/activitypub/services/activitypub/subservices/inbox.js
Comment (KrinsKumar) [reply src/middleware/packages/activitypub/services/activitypub/subservices/inbox.js]: fixed.

**Review by srosset81 — APPROVED** [review]

**Comment by srosset81** [comment]
Thanks @KrinsKumar !
Maybe you could add a [test for inboxes](https://github.com/assemblee-virtuelle/semapps/blob/master/src/middleware/tests/activitypub/inbox.test.js) in case a user doesn't exist ?

**Comment by KrinsKumar** [comment]
I have added a test, I will merge the PR once you approve the test
Here is a screenshot of the tests passing
image

**Review by srosset81 — APPROVED** [review]
Great work ! Thank you

**Pull request merged**
